### DISCUSSION TITLE
Unable connect to rabbitmq when use queue with specific letter (^M)

### DISCUSSION THREAD

**Discussion by NoraDing** [rabbitmq/rabbitmq-server#11205] [Questions]
Hi 

We got one connect exception when we use a queue including  ^M ,queue name likes 'xxxxx88^M   _xxxxxx'. 
Can you help confirm whether it's a known issue ? Or besides '^M', is there any more letter will cause this problem?

rabbitmq version:
3.8.30

Related logs:
`2024-04-21 18:19:21.004 [error] <0.26980.61> Channel error on connection <0.18911.8126> (ip1:51482 -> ip2:5672, vhost: '/', user: 'username'), channel 401:
operation basic.consume caused a channel exception not_found: no queue 'xxxxx88^M   _xxxxxx' in vhost '/'`

`2024-04-21 18:19:23.006 [error] <0.18911.8126> Error on AMQP connection <0.18911.8126> (ip1:51482 -> ip2:5672, vhost: '/', user: 'username', state: running), channel 401:
 operation channel.close caused a connection exception channel_error: "expected 'channel.open'"
2024-04-21 18:19:23.008 [info] <0.18911.8126> closing AMQP connection <0.18911.8126> (ip1:51482 -> ip1:5672, vhost: '/', user: 'username')
2024-04-21 18:19:23.009 [info] <0.10312.37> Closing all channels from connection 'ip1:51482 -> ip2:5672' because it has been closed
2024-04-21 18:19:23.242 [info] <0.14792.67> Closing all channels from connection 'ip1:49674 -> ip2:5672' because it has been closed`

Thanks

**Comment by lukebakken** [comment]
Hello, thanks for using RabbitMQ.

Version 3.8.30 is out of support. Please read the community support guidelines here - https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support

If you would like assistance with this issue, you must reproduce it with the latest version of RabbitMQ, and you must provide what we need to reproduce it. In your case, a simple application that connects, declares a queue with `^M` in the name, and then tries to consume from it.

**Comment by michaelklishin** [comment]
@NoraDing to add to what @lukebakken has said above, here is [the list of supported versions](https://www.rabbitmq.com/release-information). 

"^M" is not a letter, it seems to be a control sequence. In AMQP 0-9-1, queues **must** only use the alphanumeric part of the ASCII set.

Please provide an executable way to reproduce against `3.13.2`. We do not guess in this community and guessing as to what exactly your client may be sending can waste a lot of time.

The operation in the exception is `basic.consume`, so it should be fairly trivial to reproduce (in theory) by declaring a queue and then adding a consumer.

**Reply by lukebakken** [reply]
> In AMQP 0-9-1, queues must only use the alphanumeric part of the ASCII set

I didn't realize that's a protocol limitation. Thanks @michaelklishin

**Reply by michaelklishin** [reply]
@lukebakken we do not enforce it but control sequences are clearly a bad idea and "special" characters other than dashes, underscores are generally not recommended. We simply do not test many of those combinations and it's not a priority.
